### PR TITLE
Add support for metadata in CBS

### PR DESCRIPTION
**Pull request by michal-hudy**
**Description**

Changes proposed in this pull request:

- Added support for metadata in CBS

**Related issue(s)**

#3989

### CODE REVIEW

File: components/console-backend-service/internal/domain/assetstore/asset_converter.go [hunk 1]
```diff
@@ -23,12 +27,17 @@ func (c *assetConverter) ToGQL(item *v1alpha2.Asset) (*gqlschema.Asset, error) {
 	}
 
 	status := c.extractor.Status(item.Status.CommonAssetStatus)
+	metadata, err := c.extractMetadata(item.Spec.Metadata)
```
Comment (magicmatatjahu) [inline components/console-backend-service/internal/domain/assetstore/asset_converter.go]: Please change name of type of `extractor` from `AssetStatusExtractor` to `AssetExtractor` and put there `extractMetadata` and use it here.

File: components/console-backend-service/internal/domain/assetstore/asset_converter.go [hunk 2]
```diff
@@ -48,3 +57,16 @@ func (c *assetConverter) ToGQLs(in []*v1alpha2.Asset) ([]gqlschema.Asset, error)
 	}
 	return result, nil
 }
+
+func (*assetConverter) extractMetadata(ext *runtime.RawExtension) (map[string]interface{}, error) {
```
Comment (magicmatatjahu) [inline components/console-backend-service/internal/domain/assetstore/asset_converter.go]: Please extract this function to `extractor` service.

File: components/console-backend-service/internal/domain/assetstore/clusterasset_converter.go [hunk 3]
```diff
@@ -23,11 +27,16 @@ func (c *clusterAssetConverter) ToGQL(item *v1alpha2.ClusterAsset) (*gqlschema.C
 	}
 
 	status := c.extractor.Status(item.Status.CommonAssetStatus)
+	metadata, err := c.extractMetadata(item.Spec.Metadata)
```
Comment (magicmatatjahu) [inline components/console-backend-service/internal/domain/assetstore/clusterasset_converter.go]: Please change name of type of `extractor` from `AssetStatusExtractor` to `AssetExtractor` and put there `extractMetadata` and use it here.

File: components/console-backend-service/internal/domain/assetstore/clusterasset_converter.go [hunk 4]
```diff
@@ -47,3 +56,16 @@ func (c *clusterAssetConverter) ToGQLs(in []*v1alpha2.ClusterAsset) ([]gqlschema
 	}
 	return result, nil
 }
+
+func (*clusterAssetConverter) extractMetadata(ext *runtime.RawExtension) (map[string]interface{}, error) {
```
Comment (magicmatatjahu) [inline components/console-backend-service/internal/domain/assetstore/clusterasset_converter.go]: Please extract this function to `extractor` service.

File: docs/asset-store/06-01-asset.md [hunk 5]
```diff
@@ -91,6 +93,7 @@ This table lists all possible parameters of a given resource together with their
 | **metadata.name** |    **YES**   | Specifies the name of the CR. |
 | **metadata.namespace** |    **YES**   | Defines the Namespace in which the CR is available. |
 | **spec.source.mode** |    **YES**   | Specifies if the asset consists of one file or a set of compressed files in the ZIP or TAR formats. Use `single` for one file and `package` for a set of files. |
+| **spec.source.metadata** |    **NO**   | Specifies the set of the metadata for Asset. It must be defined in a valid YAML/JSON format. |
```
Comment (kazydek) [inline docs/asset-store/06-01-asset.md]:
```suggestion
| **spec.source.metadata** |    **NO**   | Specifies a set of metadata for the Asset. Define it in a valid YAML or JSON format. |
```

File: docs/asset-store/06-03-clusterasset.md [hunk 6]
```diff
@@ -88,6 +90,7 @@ This table lists all possible parameters of a given resource together with their
 |----------|:-------------:|------|
 | **metadata.name** |    **YES**   | Specifies the name of the CR. |
 | **spec.source.mode** |    **YES**   | Specifies if the asset consists of one file or a set of compressed files in the ZIP or TAR formats. Use `single` for one file and `package` for a set of files. |
+| **spec.source.metadata** |    **NO**   | Specifies the set of the metadata for ClusterAsset. It must be defined in a valid YAML/JSON format. |
```
Comment (kazydek) [inline docs/asset-store/06-03-clusterasset.md]:
```suggestion
| **spec.source.metadata** |    **NO**   | Specifies a set of metadata for the ClusterAsset. Define it in a valid YAML or JSON format. |
```

File: docs/headless-cms/03-01-headless-cms-in-console.md [hunk 7]
```diff
@@ -12,3 +12,11 @@ Use these labels to mark and filter assets in the Console UI:
 - **cms.kyma-project.io/view-context** specifies the location in the Console UI to render the given asset. This can be either `docs-ui` or `service-catalog`.
 - **cms.kyma-project.io/group-name** defines the group, such as `components`, under which you want to render the given asset under the `docs-ui` view in the Console UI. The value cannot include spaces.
 - **cms.kyma-project.io/order** specifies the position of the DocsTopic and ClusterDocsTopic in relation to other DocsTopics under the `docs-ui` view in the Console UI. For example, this can be `4`.
+
+## Configuration
+
+To configure how DocsTopic and ClusterDocsTopic are rendered in UI, use the following parameters:
```
Comment (kazydek) [inline docs/headless-cms/03-01-headless-cms-in-console.md]:
```suggestion
To define how DocsTopic and ClusterDocsTopic are rendered in the UI, use the following parameters:
```

File: docs/headless-cms/06-01-docstopic.md [hunk 8]
```diff
@@ -58,6 +60,7 @@ This table lists all possible parameters of a given resource together with their
 | **spec.sources.type** |    **YES**   | Specifies the type of assets included in the DocsTopic CR. |
 | **spec.sources.name** |    **YES**   | Defines an identifier of a given asset. It must be unique if there is more than one asset of a given type in a DocsTopic CR. |
 | **spec.sources.mode** |    **YES**   | Specifies if the asset consists of one file or a set of compressed files in the ZIP or TAR format. Use `single` for one file and `package` for a set of files.  |
+| **spec.sources.metadata** |    **NO**   | Specifies the set of the metadata for Asset. It must be defined in a valid YAML/JSON format. |
```
Comment (kazydek) [inline docs/headless-cms/06-01-docstopic.md]:
```suggestion
| **spec.sources.metadata** |    **NO**   | Specifies a set of metadata for the Asset. Define it in a valid YAML or JSON format. |
```

File: docs/headless-cms/06-02-clusterdocstopic.md [hunk 9]
```diff
@@ -54,6 +56,7 @@ This table lists all possible parameters of a given resource together with their
 | **spec.sources.type** |    **YES**   | Specifies the type of assets included in the ClusterDocsTopic CR. |
 | **spec.sources.name** |    **YES**   | Defines a unique identifier of a given asset. It must be unique if there is more than one asset of a given type in a ClusterDocsTopic CR. |
 | **spec.sources.mode** |    **YES**   | Specifies if the asset consists of one file or a set of compressed files in the ZIP or TAR format. Use `single` for one file and `package` for a set of files.  |
+| **spec.sources.metadata** |    **NO**   | Specifies the set of the metadata for ClusterAsset. It must be defined in a valid YAML/JSON format. |
```
Comment (kazydek) [inline docs/headless-cms/06-02-clusterdocstopic.md]:
```suggestion
| **spec.sources.metadata** |    **NO**   | Specifies a set of metadata for the ClusterAsset. Define it in a valid YAML or JSON format. |
```

File: docs/headless-cms/06-01-docstopic.md [hunk 10]
```diff
@@ -58,6 +60,7 @@ This table lists all possible parameters of a given resource together with their
 | **spec.sources.type** |    **YES**   | Specifies the type of assets included in the DocsTopic CR. |
 | **spec.sources.name** |    **YES**   | Defines an identifier of a given asset. It must be unique if there is more than one asset of a given type in a DocsTopic CR. |
 | **spec.sources.mode** |    **YES**   | Specifies if the asset consists of one file or a set of compressed files in the ZIP or TAR format. Use `single` for one file and `package` for a set of files.  |
+| **spec.sources.metadata** |    **NO**   | Specifies a set of metadata for the Asset. Define it in a valid YAML or JSON format. |
```
Comment (derberg) [inline docs/headless-cms/06-01-docstopic.md]: in each doc where we describe **spec.sources.metadata** specify for the user that one of the possible use case for metadata is to use is as a contract between source and the presentation layer, to enable UI to decide on a  source level, if something specific has to be done/modified in the source before it is rendered for the user

File: docs/headless-cms/03-01-headless-cms-in-console.md [hunk 11]
```diff
@@ -12,3 +12,11 @@ Use these labels to mark and filter assets in the Console UI:
...
 - **cms.kyma-project.io/group-name** defines the group, such as `components`, under which you want to render the given asset under the `docs-ui` view in the Console UI. The value cannot include spaces.
 - **cms.kyma-project.io/order** specifies the position of the DocsTopic and ClusterDocsTopic in relation to other DocsTopics under the `docs-ui` view in the Console UI. For example, this can be `4`.
+
+## Configuration
+
+To define how DocsTopic and ClusterDocsTopic are rendered in the UI, use the following parameters:
+
+| Parameter | Default Value | Description |
+| --------- | ------------- | ----------- |
+| **spec.sources.metadata.disableRelativeLinks** | `false` | Disables relative links when documentation is rendered. |
```
Comment (kazydek) [inline docs/headless-cms/03-01-headless-cms-in-console.md]:
```suggestion
| **spec.sources.metadata.disableRelativeLinks** | `false` | Disables relative links when documentation is rendered. It only applies to the `markdown` type of assets included in the (Cluster)DocsTopic CR. |
```

File: docs/headless-cms/06-01-docstopic.md (hunk 10 above)
Comment (kazydek) [inline docs/headless-cms/06-01-docstopic.md]:
```suggestion
| **spec.sources.metadata** |    **NO**   | Specifies a set of metadata for the Asset that you want to render, disable, or modify in the UI. Define it in a valid YAML or JSON format. |
```

Comment (kazydek) [reply docs/headless-cms/06-01-docstopic.md]: and pls copy it to every doc when this parameter is described ;)

**Pull request merged**